src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/Ordering/PasteOrdering.cs

```diff
@@ -0,0 +1,141 @@
+    /// </summary>
+    [Export(typeof(IPasteDataObjectProcessor))]
+    [Export(typeof(IPasteHandler))]
+    [AppliesTo(ProjectCapabilities.SortByDisplayOrder)]
+    [Order(OrderPrecedence)]
+    internal class PasteOrdering : IPasteHandler, IPasteDataObjectProcessor
+    {
+        public const int OrderPrecedence = 10000;
+
+        private readonly ConfiguredProject _configuredProject;
+        private readonly IProjectAccessor _accessor;
+
+        private IProjectTree _dropTarget;
+
+        private IPasteHandler _pasteHandler;
+        private IPasteDataObjectProcessor _pasteProcessor;
+
+        [ImportingConstructor]
+        public PasteOrdering(UnconfiguredProject unconfiguredProject, IProjectAccessor accessor)
+        {
+            Requires.NotNull(unconfiguredProject, nameof(unconfiguredProject));
+            Requires.NotNull(accessor, nameof(accessor));
+
+            _configuredProject = unconfiguredProject.Services.ActiveConfiguredProjectProvider.ActiveConfiguredProject;
+            _accessor = accessor;
+
+            PasteHandlers = new OrderPrecedenceImportCollection<IPasteHandler>(projectCapabilityCheckProvider: unconfiguredProject);
+            PasteProcessors = new OrderPrecedenceImportCollection<IPasteDataObjectProcessor>(projectCapabilityCheckProvider: unconfiguredProject);
+        }
+
+        [ImportMany]
+        private OrderPrecedenceImportCollection<IPasteHandler> PasteHandlers { get; set; }
+
+        [ImportMany]
+        private OrderPrecedenceImportCollection<IPasteDataObjectProcessor> PasteProcessors { get; set; }
+
+        private IPasteHandler PasteHandler
+        {
+            get
+            {
+                if (_pasteHandler == null)
+                {
+                    // Grab the paste handler that has the highest order precedence that is below PasteOrdering's order precedence. 
+                    _pasteHandler =
+                        PasteHandlers.Where(x => x.Metadata.OrderPrecedence < OrderPrecedence)
+                        .OrderByDescending(x => x.Metadata.OrderPrecedence).First().Value;
+                }
+
+                Assumes.NotNull(_pasteHandler);
+
+                return _pasteHandler;
+            }
+        }
+
+        private IPasteDataObjectProcessor PasteProcessor
+        {
+            get
+            {
+                if (_pasteProcessor == null)
+                {
+                    // Grab the paste processor that has the highest order precedence that is below PasteOrdering's order precedence. 
+                    _pasteProcessor =
+                        PasteProcessors.Where(x => x.Metadata.OrderPrecedence < OrderPrecedence)
+                        .OrderByDescending(x => x.Metadata.OrderPrecedence).First().Value;
+                }
+
+                Assumes.NotNull(_pasteProcessor);
+
+                return _pasteProcessor;
+            }
+        }
+
+        #region IPasteDataObjectProcessor
```

Comment:
🔥 as per `We avoid using regions.`?

https://github.com/dotnet/project-system/blob/master/docs/repo/coding-conventions.md#code

Reply:
yeah, sorry, no regions please

Reply:
Ok